apollo-gradle-plugin/src/main/groovy/com/apollostack/android/gradle/ApolloIRgenTask.groovy

```diff
@@ -36,48 +38,79 @@ public class ApolloIRGenTask extends NodeTask {
     variant = variantName
     config = extensionsConfig
     group = ApolloPlugin.TASK_GROUP
-    description = "Generate IR files using apollo-codgen for ${variant.capitalize()} GraphQL queries"
+    description = "Generate an IR file using apollo-codgen for ${variant.capitalize()} GraphQL queries"
     dependsOn(ApolloCodeGenInstallTask.NAME)
-    outputDir = new File("${getSchemaPath()}/generatedIR")
+
+    possibleGraphQLPaths = buildPossibleGraphQLPaths()
+    outputDir = project.file("src/$variant/graphql/generatedIR")
   }
 
   @Override
   public void exec() {
     File apolloScript = project.file(APOLLO_CODEGEN)
+    File schemaFile = userProvidedSchemaFile() ?: searchForSchemaFile()
 
     if (!apolloScript.isFile()) {
       throw new GradleException("Apollo-codegen was not found in node_modules. Please run 'gradle " +
           "${ApolloCodeGenInstallTask.NAME}")
     }
-    if (!new File("$outputDir.parent/schema.json").isFile()) {
+
+    if (!schemaFile.isFile()) {
       throw new GradleException("Couldn't find a schema file. Please ensure a valid schema.json files exists in the " +
           "sourceSet directory")
     }
 
     setScript(apolloScript)
     List<String> apolloArgs = ["generate"]
+    println "sending request with schema path ${schemaFile.absolutePath}"
     apolloArgs.addAll(getInputFiles().collect { project.file(it).absolutePath })
-    apolloArgs.addAll(["--schema", "$outputDir.parent/schema.json",
+    apolloArgs.addAll(["--schema", "${schemaFile.absolutePath}",
                        "--output", "$outputDir.absolutePath/${variant.capitalize()}API.json",
                        "--target", "json"]);
     setArgs(apolloArgs)
     super.exec()
   }
 
-  private String getSchemaPath() {
-    String schemaPath = ""
-    PatternSet patternSet = new PatternSet().include("**/schema.json")
-    for (String path : config.reverse().graphqlPath) {
+  private File userProvidedSchemaFile() {
+    List<String> schemaFiles = config.schemaFile?.findAll { it != null }
+    File schemaFile
+    if (schemaFiles) {
+      if (schemaFiles.size() > 1) {
+        throw new IllegalArgumentException("More than two schema files were specified for the build variant $variant." +
+            " Please ensure that only one schema field is specified for $variant's source sets")
+      }
+      schemaFile = project.file(schemaFile.get(0))
+    }
+    return schemaFile
+  }
+
+  private File searchForSchemaFile() {
```

Comment:
This is where you end up if a `schemaFile` wasn't provided in the sourceSet extension. 

This searches the `possibleGrapQLPaths` list which contains the default sourceSets paths: `src/sourceSet/graphql` plus any user provided `graphQLPath` extensions